src/content/en/fundamentals/media/fast-playback-with-video-preload.md

```diff
@@ -0,0 +1,578 @@
+spec) which can hang loading of resources. Note that this may also improve page
+speed if videos aren't part of your core user experience.
+
+## Link preload
+
+As [covered] in other [articles], [link preload] is a declarative fetch that
+allows you to force the browser to make a request for a resource without
+blocking the `window.onload` event. Resources loaded via `<link rel="preload">`
+are stored locally in the browser, and are effectively inert until they're
+explicitly referenced in the DOM, JavaScript, or CSS.
+
+Preload is different from prefetch in that it focuses on current navigation and
+fetches resources with priority based on their type (script, style, font,
+video, audio, etc.). It should be used to warm up the browser cache for current
+sessions.
+
+<figure>
+  <img src="/web/fundamentals/media/images/video-preload/link-preload.svg">
+</figure>
+
+### Preload full video
+
+Here's how to preload a full video  on your website so that when your
+JavaScript asks to fetch video content, it is read from disk cache as the
+resource may have already been cached by the browser. If preload request hasn't
+finished yet, a regular network fetch will happen.
+
+```
+<link rel="preload" as="video" href="https://cdn.com/small-file.mp4">
+
+<video id="video" controls></video>
+
+<script>
+  // Later on, after some condition has been met, set video source to the
+  // preloaded video URL.
+  video.src = 'https://cdn.com/small-file.mp4';
+  video.play().then(_ => {
+    // If preloaded video URL was already cached, playback started immediately.
+  });
+</script>
+```
+
+Note: I would recommend using this only for audio files or short video clips (< 5MB).
+
+Because the preloaded resource is going to be consumed by a video element in
+the example, the `as` preload link value is `video`. If it were an audio
+element, it would be `as="audio"`.
+
+### Preload first segment
+
+The example below shows how to preload the first segment of a video with `<link
+rel="preload">` and use it with Media Source Extensions. If you're not familiar
+with the MSE Javascript API, please read [MSE basics].
+
+For the sake of simplicity, let's assume the entire video has been split into
+smaller files like "file_1.webm", "file_2.webm", "file_3.webm", etc. We'll use
+MSE to feed the video element with those once the `onload` event from `<link
+rel="preload">` has been fired.
+
+```
+<link rel="preload" as="video" href="https://cdn.com/file_1.webm"
+    onload="preloadFinished()" onerror="preloadFailed()">
+
+<video id="video" controls></video>
+
+<script>
+  // This function is automatically executed when the first segment of video is
+  // preloaded successfully. We may call it explicitly as well later on.
+  function preloadFinished() {
+    const mediaSource = new MediaSource();
+    video.src = URL.createObjectURL(mediaSource);
+    mediaSource.addEventListener('sourceopen', sourceOpen);
+  }
+
+  function sourceOpen(event) {
+    URL.revokeObjectURL(video.src);
```

Comment:
Agreed. We (MSE spec) were originally going the route for autorevoking URLs IIRC (paralleling the FILE API at the time), but due to FILE API change before MSE got to REC (and srcObject not in MSE either), the push to REC reconciled the concerns by making MSE Object URLs non-auto-revoking. srcObject with MSE is indeed something that would alleviate this particular concern/pattern mess.